### PR TITLE
[Enhancement] Support insert timeout

### PR DESCRIPTION
**Pull request by wyb**
## Why I'm doing:
`insert` and `query` use the same `query_timeout` session variable to control timeout.
but they need different default values, default `query_timeout 300s` is very small for insert.

## What I'm doing:
add a new `insert_timeout` session variable for `insert/update/delete` and the default value is `14400s`.

you can also set `timeout` in `insert properties` for the current insert sql, for example
```
insert into tbl properties("timeout" = "3600") select ...
```
if `timeout` property is not set, `insert_timeout` session variable will be used.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/qe/ResultReceiver.java [hunk 1]
```diff
@@ -139,17 +141,16 @@ public RowBatch getNext(Status status) throws TException {
             LOG.warn("fetch result execution exception, finstId={}", DebugUtil.printId(finstId), e);
             if (e.getMessage().contains("time out")) {
                 // if timeout, we set error code to TIMEOUT, and it will not retry querying.
-                status.setStatus(new Status(TStatusCode.TIMEOUT,
-                        String.format("Query exceeded time limit of %d seconds",
-                                ConnectContext.get().getSessionVariable().getQueryTimeoutS())));
+                status.setStatus(new Status(TStatusCode.TIMEOUT, ErrorCode.ERR_TIMEOUT.formatErrorMsg(
+                        "Query", timeoutMs / 1000, "please increase the 'query_timeout' session variable")));
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/qe/ResultReceiver.java:145]: query_timeout only?

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -2689,4 +2693,28 @@ public void addFinishedQueryDetail() {
...
+        if (parsedStmt instanceof CreateTableAsSelectStmt) {
+            InsertStmt insertStmt = ((CreateTableAsSelectStmt) parsedStmt).getInsertStmt();
+            int timeout = insertStmt.getTimeout();
+            return timeout != -1 ? timeout : context.getSessionVariable().getInsertTimeoutS();
+        }
+        return context.getSessionVariable().getQueryTimeoutS();
+    }
+
+    public String getExecType() {
+        if (parsedStmt instanceof InsertStmt || parsedStmt instanceof CreateTableAsSelectStmt) {
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:2710]: Why not implement it as a method in StatementBase?
And provide a StmtExecType class with DML_INSERT, DML_UPDATE, DML_DELETE, DCL_QUERY

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 3]
```diff
@@ -2689,4 +2693,28 @@ public void addFinishedQueryDetail() {
         QueryDetailQueue.addQueryDetail(queryDetail);
     }
 
+    public int getExecTimeout() {
+        if (parsedStmt instanceof DmlStmt) {
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:2697]: Is it better to provide a abstract getTimeout method in StatementBase, with some sub-classes implementing it?

Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java]: done

File: fe/fe-core/src/main/java/com/starrocks/qe/ResultReceiver.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/ResultReceiver.java:145]: yes, only query uses this.

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 4]
```diff
@@ -1303,6 +1304,7 @@ private void handleAnalyzeStmt() throws IOException {
                 // will print warning log if timeout, so we update timeout temporarily to avoid
                 // warning log
                 context.getSessionVariable().setQueryTimeoutS((int) Config.statistic_collect_query_timeout);
+                context.getSessionVariable().setInsertTimeoutS((int) Config.statistic_collect_query_timeout);
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1326]: analyze job has both query and insert sql, so set query_timeout and insert_timeout

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 5]
```diff
@@ -191,6 +191,7 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     public static final String MAX_PARALLEL_SCAN_INSTANCE_NUM = "max_parallel_scan_instance_num";
     public static final String ENABLE_INSERT_STRICT = "enable_insert_strict";
     public static final String INSERT_MAX_FILTER_RATIO = "insert_max_filter_ratio";
+    public static final String INSERT_TIMEOUT = "insert_timeout";
```
Comment (meegoo) [inline fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:194]: If the user has set query_timeout, how to maintain compatibility when the user upgrades?

**Review by jaogoy — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 6]
```diff
@@ -1154,6 +1155,9 @@ public static MaterializedViewRewriteMode parse(String str) {
     @VariableMgr.VarAttr(name = INSERT_MAX_FILTER_RATIO)
     private double insertMaxFilterRatio = 0;
 
+    @VariableMgr.VarAttr(name = INSERT_TIMEOUT)
+    private int insertTimeoutS = 14400;
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:1159]: the value is same as broker load default job timeout

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 5 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:194]: add `insert_timeout` session variable is to make it easier for users to change.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java:1168]: update pipe/mv default timeout same as insert timeout

**Review by LiShuMing — DISMISSED** [review]

**Review by meegoo — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Review by meegoo — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java [hunk 7]
```diff
@@ -910,7 +910,8 @@ public RowBatch getNext() throws Exception {
                         copyStatus.getErrorMsg().equals(FeConstants.BACKEND_NODE_NOT_FOUND_ERROR)) {
                     ec = InternalErrorCode.CANCEL_NODE_NOT_ALIVE_ERR;
                 } else if (copyStatus.isTimeout()) {
-                    ErrorReport.reportTimeoutException(ErrorCode.ERR_QUERY_TIMEOUT, errMsg);
+                    ErrorReport.reportTimeoutException(
+                            ErrorCode.ERR_TIMEOUT, "Query", jobSpec.getQueryOptions().query_timeout, errMsg);
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:914]: Just query_timeout or depending on isExecLoadType()?

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 8]
```diff
@@ -2342,7 +2362,8 @@ public void handleDMLStmt(ExecPlan execPlan, DmlStmt stmt) throws Exception {
                                         "or set parallel_fragment_exec_instance_num to a lower value in session variable");
                     } else {
                         ErrorReport.reportTimeoutException(ErrorCode.ERR_TIMEOUT, getExecType(), timeout,
-                                "please increase the 'insert_timeout' session variable and retry");
+                                String.format("please increase the '%s' session variable and retry",
+                                        SessionVariable.INSERT_TIMEOUT));
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:2366]: Does the timeout depend on isExecLoadType()

**Review by jaogoy — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java (hunk 7 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/DefaultCoordinator.java:914]: Only query is used here, 
and `TQueryOptions.query_timeout` is used uniformly for query and insert.

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java (hunk 8 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:2366]: only insert is used here.

**Pull request merged**